src/gamma_inc.jl

```diff
@@ -0,0 +1,599 @@
+b1=[.386325038602125E-02 , .506042559238939E-01 , .283344278023803E+00 , .780110511677243E+00]
+a2=[.669564126155663E-03 , .413359788442192E-02]
+b2=[-.421924263980656E-03 , .650837693041777E-02 , .682034997401259E-01 , .339173452092224E+00 , .810647620703045E+00]
+a3=[.810586158563431E-03 , .649434157619770E-03]
+b3=[-.632276587352120E-03 , .905375887385478E-02 , .906610359762969E-01 , .406288930253881E+00 , .894800593794972E+00]
+a4=[-.105014537920131E-03 , -.861888301199388E-03]
+b4=[.322609381345173E-01 , .178295773562970E+00 , .591353097931237E+00 , .103151890792185E+01]
+a5=[-.435211415445014E-03 , -.336806989710598E-03]
+b5=[.178716720452422E+00 , .600380376956324E+00 , .108515217314415E+01]
+a6=[-.182503596367782E-03 , .531279816209452E-03]
+b6=[.345608222411837E+00 , .770341682526774E+00]
+a7=[.443219646726422E-03 , .344430064306926E-03]
+b7=[.821824741357866E+00 , .115029088777769E+01]
+a8=[.878371203603888E-03 , -.686013280418038E-03]
+
+#----------------COEFFICIENTS FOR TEMME EXPANSION------------------
+
+d00 = -.333333333333333E+00
+d0=[.833333333333333E-01 , -.148148148148148E-01 , .115740740740741E-02 , .352733686067019E-03 , -.178755144032922E-03 , .391926317852244E-04]
+d10 = -.185185185185185E-02
+d1=[-.347222222222222E-02 , .264550264550265E-02 , -.990226337448560E-03 , .205761316872428E-03]
+d20 = .413359788359788E-02
+d2=[-.268132716049383E-02 , .771604938271605E-03]
+d30 = .649434156378601E-03
+d3=[.229472093621399E-03 , -.469189494395256E-03]
+d40 = -.861888290916712E-03
+d4=[.784039221720067E-03]
+d50 = -.336798553366358E-03
+d5=[-.697281375836586E-04]
+d60 = .531307936463992E-03
+d6=[-.592166437353694E-03]
+d70 = .344367606892378E-03
+d80 = -.652623918595309E-03
+
+# The kernel of log1pmx
+# Accuracy within ~2ulps for -0.227 < x < 0.315
+function _log1pmx_ker(x::Float64)
+    r = x/(x+2.0)
+    t = r*r
+    w = @horner(t,
+                6.66666666666666667e-1, # 2/3
+                4.00000000000000000e-1, # 2/5
+                2.85714285714285714e-1, # 2/7
+                2.22222222222222222e-1, # 2/9
+                1.81818181818181818e-1, # 2/11
+                1.53846153846153846e-1, # 2/13
+                1.33333333333333333e-1, # 2/15
+                1.17647058823529412e-1) # 2/17
+    hxsq = 0.5*x*x
+    r*(hxsq+w*t)-hxsq
+end
+"""
+    logmxp1(x::Float64)
+Return `log(x) - x + 1` carefully evaluated.
+"""
+function logmxp1(x::Float64)
+    if x <= 0.3
+        return (log(x) + 1.0) - x
+    elseif x <= 0.4
+        u = (x-0.375)/0.375
+        return _log1pmx_ker(u) - 3.55829253011726237e-1 + 0.625*u
+    elseif x <= 0.6
+        u = 2.0*(x-0.5)
+        return _log1pmx_ker(u) - 1.93147180559945309e-1 + 0.5*u
+    else
+        return log1pmx(x - 1.0)
+    end
+end
+
+"""
+   rgamma1pm1(a)
+
+   Computation of 1/Gamma(a+1) - 1 for -0.5<=a<=1.5 : ``1/\\Gamma (a+1) - 1``
+"""
+function rgamma1pm1(a::Float64)
+    t=a
+    d = a - 0.5
+    if d > 0.0
+        t = d - 0.5
+    end
```

Comment:
since `d` is only used as a test for the condition `d > 0.0`, maybe get rid of it and write:
```
rangereduce = a > 0.5
t = rangereduce ? a - 1 : a   # -0.5 <= t <= 0.5
```

Reply:
also maybe add a comment that it is making use of the relation `gamma(a+1) = a*gamma(a)`